### PR TITLE
feat: add loading and error component support to LingoProviderWrapper

### PR DESCRIPTION
**Pull request by Agarwalchetan**
This PR enhances the [LingoProviderWrapper](cci:1://file:///d:/Codes/Code/Open%20Source/Lingo.dev/lingo.dev/packages/react/src/client/provider.tsx:143:0-197:1) component in the React package by adding optional `loadingComponent` and `errorComponent` props, enabling developers to provide custom UI for loading and error states instead of the default `null` behavior.

This resolves two TODO comments in the codebase that identified the need for better loading and error state handling.

### Core Implementation
- **Added `loadingComponent` prop**: Optional component to display while the dictionary is loading
- **Added `errorComponent` prop**: Optional component to display when dictionary loading fails
- **Refactored state management**: Changed from simple `D | null` to discriminated union with `loading`, `loaded`, and `error` states
- **Type-safe error handling**: Automatically converts any thrown value to an `Error` object
- **Removed TODO comments**: Both TODOs in [provider.tsx](cci:7://file:///d:/Codes/Code/Open%20Source/Lingo.dev/lingo.dev/packages/react/src/client/provider.tsx:0:0-0:0) are now resolved

### Testing
- Added 5 comprehensive test cases covering:
  - Custom loading component rendering
  - Custom error component with error message display
  - Non-Error object conversion to Error
  - State transition verification
  - Backward compatibility
- All 77 tests in the React package pass 

##  Benefits

-  **Better UX**: Developers can now show meaningful loading states instead of blank screens
-  **Proper error handling**: Custom error UI with access to error details
-  **Backward compatible**: Maintains existing `null` behavior when props are not provided
-  **Type-safe**: Full TypeScript support with proper error typing
-  **Well documented**: Comprehensive JSDoc with usage examples

##  Example Usage

```tsx
import { LingoProviderWrapper } from "lingo.dev/react/client";

function App() {
  return (
    <LingoProviderWrapper
      loadDictionary={loadDictionary}
      loadingComponent={
        <div className="loading-spinner">
          Loading translations...
        </div>
      }
      errorComponent={({ error }) => (
        <div className="error-message">
          Failed to load translations: {error.message}
        </div>
      )}
    >
      <YourApp />
    </LingoProviderWrapper>
  );
}
```

##  Checklist

- [x] Linked to existing issue (addresses TODOs in codebase)
- [x] Added comprehensive tests (5 new test cases, all passing)
- [x] Added changeset (`pnpm new`)
- [x] All tests pass (`pnpm turbo test`)
- [x] Code formatted (`pnpm format`)
- [x] Format check passes (`pnpm format:check`)
- [x] Backward compatible (no breaking changes)
- [x] Documentation included (JSDoc with examples)

##  Code Quality

- **Surgical PR**: Single, focused objective
- **Elegant implementation**: Uses TypeScript discriminated unions for type-safe state management
- **Comprehensive testing**: Tests cover both positive and negative paths, including edge cases
- **Zero breaking changes**: Fully backward compatible with existing implementations

##  Test Results

```
Test Files  18 passed (18)
Tests       77 passed (77)
Duration    22.99s
```

##  Before/After

**Before:**
```tsx
// Loading state: renders null (blank screen)
// Error state: renders null (blank screen, error only in console)
```

**After:**
```tsx
// Loading state: renders custom loading component (optional)
// Error state: renders custom error component with error details (optional)
// Default: maintains backward compatible null behavior
```

### CODE REVIEW

File: packages/react/src/client/provider.spec.tsx [hunk 1]
```diff
@@ -84,5 +84,103 @@ describe("client/provider", () => {
...
+      const errorEl = await findByTestId("error");
+      expect(errorEl.textContent).toBe("string error");
+    });
+
+    it("transitions from loading to loaded to error states correctly", async () => {
+      const loadDictionary = vi
+        .fn()
+        .mockResolvedValue({ locale: "en", files: {} });
+
+      const { findByTestId, rerender } = render(
```
[the remark this answers is not in the document]
Comment (Agarwalchetan) [reply packages/react/src/client/provider.spec.tsx:205]: Improved Old Problems

**Comment by Agarwalchetan** [comment]
Again Check this Request
